src/JuliennedArrays.jl

```diff
@@ -236,19 +325,37 @@ julia> slices
  [0, 2]
  [3, 4]
 ```
+
+Will throw an error if you try to align slices with different axes.
+Use `@inbounds` to skip this check.
+
+```jldoctest align_bools
+julia> unequal = [[1], [1, 2]];
+
+julia> Align(unequal, False(), True())
+ERROR: ArgumentError: Axes of [1, 2] does not match the axes of the first slice: (Base.OneTo(1),)
+[...]
+
+julia> (array -> @inbounds Align(array, False(), True()))(unequal) # broken
```

Comment:
This works locally but not in the doctest. Maybe something to do with inlining??

Reply:
Oh wait, I think it's because of tests run with the `check-bounds=yes` option